Controls/Mail/Mailman.c

```diff
@@ -11,8 +13,21 @@
  * Once the letter is deposited, signal the semaphore (effectively putting up the flag).
  * Print "Finished!" at the end.
  */
+//extern char* mailbox;
 void depositLetter(void) {
 	OS_ERR err; // Make sure to check for errors and print the error code if not OS_ERR_NONE
+	//int i;
+	//char max=random()%255;//different lengths at diff times
+	for(int i=0;i<255;i++){
+		mailbox[i]=64+rand()%63;
+	}
+	mailbox[256]=0;//null at end
+	OSSemPost(&MailboxFlag_Sem4, OS_OPT_POST_NONE, &err);//post +1 mail in box
+		if(err!=OS_ERR_NONE){
+			printf("Error from OSSemPost Mailman, depositLetter");
+			return;
+		}
```

Comment:
Indentation, also you could try adding the error code of the os error in the print for any os functions you call.